### ISSUE TITLE
Allow creation of new file # new file: /Users/Acme/Desktop/ProjectName/FormulaPopup.swift? y

### ISSUE THREAD

**Issue by SeventyKnots** [paul-gauthier/aider#107]
Hey,

Not sure if this a bug or something new entirely. I'm completely new at this so please forgive my ignorance. If it's not an issue feel free to delete. However it doesn't have an issue when you are exporting .html or .py files alike.

Error
When generating code (in this case SwiftUI) and it wants to save the file to the location it creates "# new file:" as part of a new folder structure.  "Allow creation of new file # new file: /Users/Acme/Desktop/ProjectName/FormulaPopup.swift? y"

Expected Results
When generating the code it outputs into the directory that you are currently working from i.e. the project. 

Actual Results
Additional folder structure is created with "# new file:" as the name

Cheers!
Jeremy

**Comment by paul-gauthier** [comment]
Thanks for trying aider and reporting this issue.

It sounds like GPT has tried to create a file with a wildly inappropriate filename. This is one of the reasons aider asks for confirmation before creating new files for GPT. Are you using GPT-3.5?

If this happens, you could say "no" to the file creation confirmation, then tell GPT the filename you want it to use and see if it re-sends the code with the right filename. Or you could use `/add` to add the correct filename yourself and then GPT will probably figure out to edit that empty file if you ask it for the code change again.

You might also find this FAQ entry helpful:
https://aider.chat/docs/faq.html#aider-isnt-editing-my-files

**Comment by SeventyKnots** [comment]
Yeah. Dunno. I'll try what you suggested. The previous file that was created was called contentView.swift. Same results. I'm using GPT-4. Sorry I should have said that. I hadn't thought of saying no, then /add the file. I'll give that a go and see if it's the same results or not. Thanks for the advise.

**Comment by SeventyKnots** [comment]
I have had partial success, again I was using GPT-4, but I had to tell it where the path was to the folder where I wanted it to save files. You were correct in some times it would fail but I just tried it again and it seems to work. Also /add file also works. But the best is teaching the AI the actual location at the very beginning of requests. It does seem to do well with nested folders i.e. if you MVC architecture and using folders to help visually see the areas...

But It works hope that helps anyone having similar issues.
